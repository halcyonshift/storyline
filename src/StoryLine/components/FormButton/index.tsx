import { Box, Button } from '@mui/material'
import { FormButtonProps } from './types'

const FormButton = ({ label }: FormButtonProps) => (
    <Box className='text-right'>
        <Button type='submit' variant='contained'>
            {label}
        </Button>
    </Box>
)

export default FormButton
