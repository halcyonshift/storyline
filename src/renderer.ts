import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
