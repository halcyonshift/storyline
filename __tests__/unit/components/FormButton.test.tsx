import renderer from 'react-test-renderer'
import FormButton from '@sl/components/FormButton'
import { render, screen } from '../test-utils'

describe('<FormButton />', () => {
    it('has the correct label', async () => {
        render(<FormButton label='Submit' />)
        expect(screen.getByText('Submit')).toBeTruthy()
    })

    it('renders correctly', () => {
        const tree = renderer.create(<FormButton label='Submit' />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
