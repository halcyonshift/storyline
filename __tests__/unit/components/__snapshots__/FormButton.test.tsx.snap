// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormButton /> renders correctly 1`] = `
<div
  className="text-right MuiBox-root css-0"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    Submit
  </button>
</div>
`;
